### PR TITLE
ES:14 Added handling for setSize with Alletra support

### PR DESCRIPTION
**Pull request by jyotsnalothe**
Can you please review the changes for setSize for Alletra :
@sneharai4 

Please find attached test results for testing CPG creation on 3PAR/Primera/Alletra 9K
[Test CPG changes for Alletra 9K.docx](https://github.com/hpe-storage/python-3parclient/files/6862180/Test.CPG.changes.for.Alletra.9K.docx)

### CODE REVIEW

**Comment by jyotsnalothe** [comment]
Update the repo for Rebranding changes, please have a look at it. All the UTs are passing after the changes. PFA the test results.
[python_3parclient_unittests.txt](https://github.com/hpe-storage/python-3parclient/files/6952637/python_3parclient_unittests.txt)

@sneharai4

File: hpe3parclient/client.py [hunk 1]
```diff
@@ -2124,7 +2123,16 @@ def createCPG(self, name, optional=None):
         """
         info = {'name': name}
         if optional:
-            info = self._mergeDict(info, optional)
+           if self.primera_supported:
+               for key, value in dict(optional).items():
+                   if key == 'LDLayout':
+                       ldlayout=value
```
Comment (sneharai4) [inline hpe3parclient/client.py:2129]: Put spaces round =

File: hpe3parclient/client.py [hunk 2]
```diff
@@ -2124,7 +2123,16 @@ def createCPG(self, name, optional=None):
...
+                       for keys,val in dict(ldlayout).items():
```
Comment (sneharai4) [inline hpe3parclient/client.py:2130]: Put space after keys,

File: hpe3parclient/client.py [hunk 3]
```diff
@@ -2124,7 +2123,16 @@ def createCPG(self, name, optional=None):
...
+                           if keys == 'setSize':
```
Comment (sneharai4) [inline hpe3parclient/client.py:2131]: can we change the condition to below?
if keys == 'setSize' || (keys == 'RAIDType' and ldlayout.get('RAIDType')==1):

**Review by sneharai4 — CHANGES_REQUESTED** [review]
@jyotsnalothe  Please check the review comments

File: README.rst [hunk 4]
```diff
@@ -28,6 +28,7 @@ Requirements
 ============
 This branch requires 3.1.3 version MU1 or later of the HPE 3PAR firmware
 This branch requires 4.0.0 version of the HPE Primera firmware
```
Comment (sneharai4) [inline README.rst:30]: Change the version to 4.3.1 instead of 4.0.0

File: README.rst [hunk 5]
```diff
@@ -28,6 +28,7 @@ Requirements
...
+This branch requires 9.4.0 version of the HPE Alletra firmware
```
Comment (sneharai4) [inline README.rst:31]: Change the version to 9.3.0 instead of 9.4.0. Also add 9000 after Alletra

File: hpe3parclient/client.py (hunk 1 above)
Comment (jyotsnalothe) [reply hpe3parclient/client.py]: Addressed review comments

File: hpe3parclient/client.py (hunk 2 above)
Comment (jyotsnalothe) [reply hpe3parclient/client.py]: Addressed review comments

File: hpe3parclient/client.py (hunk 3 above)
Comment (jyotsnalothe) [reply hpe3parclient/client.py]: Addressed review comments

File: README.rst (hunk 4 above)
Comment (jyotsnalothe) [reply README.rst]: Addressed review comments

File: README.rst (hunk 5 above)
Comment (jyotsnalothe) [reply README.rst]: Addressed review comments

**Review by sneharai4 — APPROVED** [review]

**Review by sneharai4 — APPROVED** [review]

**Pull request merged**